docs/source/tutorial/data-source.md

```diff
@@ -3,6 +3,424 @@ title: "2. Hook up your data sources"
+```
+
+The `https://mvrp.herokuapp.com/api/` endpoint is a simple REST API that returns data for cars. Furthermore, the `MvrpAPI` class implementation in the code above contains a `getAllCars` and `getACar` functions that wrap convenience methods provided by the `RESTDataSource` class for performing HTTP requests.
+
+In this example, the built-in `get` method used is responsible for `GET` requests. The `RESTDataSource` class provides other methods like `post`, `delete`, and `patch` for `POST`, `DELETE`, and `PATCH` requests respectively.
+
+Now that you have an understanding of how data sources work, let's hook it up for our tutorial app.
+
+Create a new `datasources` folder inside the `src` directory. This folder will contain our data source files. Now, create `launch.js` within the `datasources` directory.
+
+The REST API endpoint we'll use for our app is `https://api.spacexdata.com/v2/`. Go ahead and add the endpoint as the base URL as shown in the code below:
+
+_src/datasources/launch.js_
+
+```js
+const { RESTDataSource } = require('apollo-datasource-rest');
+
+class LaunchAPI extends RESTDataSource {
+  constructor() {
+    super();
+    this.baseURL = 'https://api.spacexdata.com/v2/';
+  }
+}
+
+module.exports = LaunchAPI;
+```
+
+In the code above, we required the `RESTDataSource` class and extended it with our custom `LaunchAPI` class. We then set up the base URL in the class constructor to the endpoint that fetches the data needed for our application.
+
+The next step is to add methods to the `LaunchAPI` class that corresponds to the type of queries our UI will fetch from the server. According to our app specifications, we'll need to get all the launches, and get a specific launch. So, let's take care of the former immediately.
+
+_src/datasources/launch.js_
+
+```js
+...
+async getAllLaunches() {
+  const res = await this.get('launches');
+
+  return res && res.length ? res.map(launch => {
+    return {
+      id: launch.flight_number || 0,
+      cursor: `${launch.launch_date_unix}`,
+      mission: {
+        name: launch.mission_name,
+        missionPatch: launch.links.mission_patch_small
+      },
+      year: launch.launch_year,
+      rocket: {
+        id: launch.rocket.rocket_id,
+        name: launch.rocket.rocket_name,
+        type: launch.rocket.rocket_type,
+      },
+      launchSuccess: launch.launch_success,
+    };
+  }) : [];
+}
+```
+
+In the code above, `this.get('launches')`, makes a `GET` request to `https://api.spacexdata.com/v2/launches` and stores the returned data in the `res` variable. If the `res` variable is not empty, then the `getAllLaunches` method returns an object that corresponds with the schema fields of the `Launch` schema type, else, an empty array is returned.
+
+Let's refactor the `getAllLaunches` method to be a lot cleaner and concise. Copy the `launchReducer` method below and add to the file. Now, refactor the `getAllLaunches` method to use the `launchReducer` method as shown below:
```

Comment:
I see where you're going with this, but I think we should teach them the `launchReducer` first instead of dragging out the example/